contrib/format-ltsv/src/test/java/org/apache/drill/exec/store/ltsv/TestLTSVRecordReader.java

```diff
@@ -17,84 +17,190 @@
+      .addNullable("host",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("forwardedfor",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("req",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("status",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("size",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("referer",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("ua",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("reqtime",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("apptime",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("vhost",  TypeProtos.MinorType.VARCHAR)
+      .buildSchema();
+
+    RowSet expected = new RowSetBuilder(client.allocator(), expectedSchema)
+      .addRow("xxx.xxx.xxx.xxx", "-", "GET /v1/yyy HTTP/1.1", "200", "412", "-", "Java/1.8.0_201", "3.580", "3.580", "api.example.com")
+      .build();
+
+    new RowSetComparison(expected).verifyAndClearAll(results);
   }
 
   @Test
   public void testSkipEmptyLines() throws Exception {
-    assertEquals(2, queryBuilder().sql("SELECT * FROM cp.`emptylines.ltsv`").run().recordCount());
+    assertEquals(2, queryBuilder().sql("SELECT * FROM cp.`ltsv/emptylines.ltsv`").run().recordCount());
   }
 
   @Test
   public void testReadException() throws Exception {
     try {
-      run("SELECT * FROM cp.`invalid.ltsv`");
+      run("SELECT * FROM cp.`ltsv/invalid.ltsv`");
       fail();
     } catch (UserException e) {
       assertEquals(UserBitShared.DrillPBError.ErrorType.DATA_READ, e.getErrorType());
-      assertTrue(e.getMessage().contains("Failure while reading messages from /invalid.ltsv. Record reader was at record: 1"));
+      assertTrue(e.getMessage().contains("Invalid LTSV format at line 1: time:30/Nov/2016:00:55:08 +0900"));
     }
   }
 
+  @Test
+  public void testSerDe() throws Exception {
+    String sql = "SELECT COUNT(*) as cnt FROM cp.`ltsv/simple.ltsv`";
+    String plan = queryBuilder().sql(sql).explainJson();
+    long cnt = queryBuilder().physical(plan).singletonLong();
+    assertEquals("Counts should match",2L, cnt);
+  }
+
+  @Test
+  public void testSelectColumnWithCompressedFile() throws Exception {
+    generateCompressedFile("ltsv/simple.ltsv", "zip", "ltsv/simple.ltsv.zip" );
+
+    String sql = "SELECT ua, reqtime FROM dfs.`ltsv/simple.ltsv.zip`";
+
+    QueryBuilder q = client.queryBuilder().sql(sql);
+    RowSet results = q.rowSet();
+
+    TupleMetadata expectedSchema = new SchemaBuilder()
+      .addNullable("ua",  TypeProtos.MinorType.VARCHAR)
+      .addNullable("reqtime",  TypeProtos.MinorType.VARCHAR)
+      .buildSchema();
+
+    RowSet expected = new RowSetBuilder(client.allocator(), expectedSchema)
+      .addRow("Java/1.8.0_131", "2.532")
+      .addRow("Java/1.8.0_201", "3.580")
+      .build();
+
+    new RowSetComparison(expected).verifyAndClearAll(results);
+  }
+
+  private void generateCompressedFile(String fileName, String codecName, String outFileName) throws IOException {
```

Comment:
You have already repeated this code in several test, let's move code to the method `ClusterFixture` and reuse it. Also clean usage in other tests.

Reply:
Moved to `ClusterTest` class and removed code from other plugin tests.